### ISSUE TITLE
Add support for equation numbering and cross-referencing

### ISSUE THREAD

**Issue by siddjain** [asciidoctor/asciidoctor#3518]
It seems asciidoc has no support for equation numbering and cross-referencing? please see these threads:
https://github.com/asciidoctor/asciidoctor-mathematical/issues/37#issuecomment-568772905
could support for this be added please?

**Comment by mojavelinux** [comment]
Equation numbering is handled by MathJax itself. It can be enabled by setting the `eqnums` document attribute. Other equations can then refer to these numbers.

You can also give the block and ID so you can reference it from AsciiDoc content.

```
:stem: latexmath
:eqnums:

[latexmath#eq-1,reftext=Equation 1]
++++
\begin{equation}
y = x^2
\end{equation}
++++

see <<eq-1>>.
```

**Comment by mojavelinux** [comment]
See https://docs.mathjax.org/en/v2.7-latest/tex.html#automatic-equation-numbering for more info.

**Comment by siddjain** [comment]
Thanks. Much obliged. This almost works for me except for 2 things:

  1.  The equation number appears on the left of the equation whereas standard practice is to have the label on the right
  2.  I would like equation numbers to be prefixed by the chapter number. Right now my equations are numbered 1, 2, 3 and so on. I would like them to be numbered 3.1, 3.2, 3.3 and so on in chapter 3

Is there any way to fix above 2 issues?

________________________________

**Comment by mojavelinux** [comment]
The equation number should be appearing on the right. Here's the output I see:

![screenshot](https://user-images.githubusercontent.com/79351/71427501-9a99f980-2676-11ea-9f00-2e48453fe6cd.png)

MathJax isn't aware of the chapters in the AsciiDoc document, so it's going to number the equations straight. To change that, you'd need to load an extension to MathJax that changes how the number is done. So no, not without a customization.

**Comment by siddjain** [comment]
@mojavelinux, is the output you have pasted the result of running asciidoctor-pdf. Below is my asciidoc

```
:stem: latexmath
:eqnums:
:chapter: 1

[latexmath#formula, id="formula", reftext={chapter}.{counter:equation}]
++++
\begin{equation}
y = x^2
\end{equation}
++++

See <<formula>>
```

and when I process it through asciidoctor-pdf I get

![image](https://user-images.githubusercontent.com/758321/75402738-7fd44300-58ba-11ea-90da-76bc93c7a34c.png)

The equation number appears on the left. I am using the docker image `asciidoctor/docker-asciidoctor`

```
asciidoctor/docker-asciidoctor                                                                         latest                        50b355abeb79        3 months ago        419MB
```

Any chance you could add ability for user to customize the equation number so that it can be numbered `7.1` for example. `7` is the chapter number.

**Comment by siddjain** [comment]
adding a note to help readers later on. the left-side equation numbering got fixed when i upgraded to asciidoctor-pdf v1.5.3. earlier i was using beta7 version of 1.5.0

**Comment by RoxvilleRooster** [comment]
I am testing equation references for Antora as described by @mojavelinux in https://github.com/asciidoctor/asciidoctor/issues/3518#issuecomment-568813204

The linking aspects work fine in Antora but the key piece, the equation reference number, is not rendered in the Antora output.
The equation reference number is rendered well in the Asciidoctor HTML (printed on the right)

```
[latexmath#eq1_sft_ca_cap,reftext=Equation 1]
++++
\begin{equation}
{MtM}_{L} = Sign \times {FX}_{currency\rightarrow DomesticCurrency} \times \left\{ \begin{matrix}
{quantity \times security\Pr ice,} & {L \in L_{SECURITY}^{NS}} \\
{cashMtM,} & {L \in L_{CASH}^{NS}} \\
\end{matrix} \right.
\end{equation}
++++
```

and the link to this is

```
where _MtM~L~_ is defined in <<#eq1_sft_ca_cap>> ...
```

**Comment by mojavelinux** [comment]
Please direct Antora questions to the Antora community chat at https://antora.zulipchat.com. The Asciidoctor issue tracker is not the appropriate place to be discussing Antora behavior.

**Comment by vishalkrsinha** [comment]
> Equation numbering is handled by MathJax itself. It can be enabled by setting the `eqnums` document attribute. Other equations can then refer to these numbers.
> 
> You can also give the block and ID so you can reference it from AsciiDoc content.
> 
> ```
> :stem: latexmath
> :eqnums:
> 
> [latexmath#eq-1,reftext=Equation 1]
> ++++
> \begin{equation}
> y = x^2
> \end{equation}
> ++++
> 
> see <<eq-1>>.
> ```

Thanks for the solution. The example works fine for cross-referencing the equation in 'asciidoctor-pdf'.
 
One query though, we see the 'reftext=Equation 1' here is the static test we are using to link. Instead of using static reftext, can we use dynamic equation number to link something like 'reftext={dynamic equation number}'?

PS: I have set ':eqnums: AMS' for automatic equation number generation.

**Comment by plevold** [comment]
@vishalkrsinha I'm currently using the following template with a counter to automatically generate equation numbers:
```asciidoctor
[stem#eq-<some label>,reftext=Equation ({counter:eqs})]
++++
<Equation here>
++++
```

Complete example:
```asciidoctor
= My Report
:sectnums:
:stem: latexmath
:eqnums: all

== Theory

[stem#eq-ode,reftext=Equation ({counter:eqs})]
++++
\mathbf{M}(t)\mathbf{\ddot{q}}(t) + \mathbf{C}(t)\mathbf{\dot{q}}(t) + \mathbf{K}(t)\mathbf{q}(t)
= \mathbf{F}(\mathbf{q}, \mathbf{\dot{q}}, t)
++++

See <<eq-ode>> for definition.

[stem#eq-emc,reftext=Equation ({counter:eqs})]
++++
E = mc^2
++++

<<eq-emc>> is another equation.
```

This should give you something like this:
![image](https://user-images.githubusercontent.com/10157556/140710079-bd971469-0108-4bc2-ac54-3a1049d6ada7.png)

You can modify the `reftext` to get the format you wish like `Eq. ({counter:eqs})` for example.

There's at least two downsides to this:
- Section numbering (for example _Equation (1.2)_) is not possible afaik.
- If you later decide to change the reftext format you have to change every single equation in your document. Maybe someone with better asciidoctor knowlege than me can figure out a way to define the format once in the document header...

**Comment by wimcnice** [comment]
Equation numbering works nice, thank you. But how can I align the equation to the left in place of center? Thanks.

**Comment by jesustorresdev** [comment]
I don't know why, but I don't see the equation number neither right nor left.

**Comment by mojavelinux** [comment]
Please direct usage questions to the project chat at https://chat.asciidoctor.org. See you there!
